pulp_python/app/tasks/publish.py

```diff
@@ -117,7 +117,9 @@ def find_artifact():
         project_dir = '{simple_dir}{name}/'.format(simple_dir=simple_dir, name=canonical_name)
         os.mkdir(project_dir)
 
-        packages = python_models.PythonPackageContent.objects.filter(name=name)
+        packages = python_models.PythonPackageContent.objects.filter(
+            name=name, pk__in=publication.repository_version.content
+        )
```

Comment:
+1

Could we also include the publishing optimization I mentioned a few weeks back?

Reply:
Yes, I'll add the work I've done on optimizing the publish.  I used your PR as inspiration, but I had to change it up a bit, however the end result is quite fast.  Still need to do more testing to find the quantitative numbers, but my initial results look very promising.